statemachine/src/main/java/com/amazonaws/stepfunctions/cloudformation/statemachine/ResourceModelUtils.java

```diff
@@ -24,8 +29,36 @@ public static void updateModelFromResult(final ResourceModel model, final Create
         model.setArn(result.getStateMachineArn());
     }
 
-    public static void updateModelFromResult(final ResourceModel model, final DescribeStateMachineResult result) {
-        model.setName(result.getName());
+    /**
+     * Generates a resource model containing the state machine resource's properties
```

Comment:
Is this supposed to contain all the fields in DescribeStateMachine's result? There are some fields missing.

Reply:
Reading this, it sounds like it doesn't need to, just the values in the resource schema. Kind of differs with @aygold92 's comment here https://github.com/aws-cloudformation/aws-cloudformation-resource-providers-stepfunctions/pull/22#discussion_r728282546

https://docs.aws.amazon.com/cloudformation-cli/latest/userguide/resource-type-test-contract.html#resource-type-test-contract-read

Reply:
From the [handler contract](https://docs.aws.amazon.com/cloudformation-cli/latest/userguide/resource-type-test-contract.html#resource-type-test-contract-read):

> The READ handler is supposed to return the you need to populate all values that are in the resource model

Reply:
This is my understanding as well, it would just be the properties defined in the resource schema. 

Values returned from [DescribeStateMachine](https://docs.aws.amazon.com/step-functions/latest/apireference/API_DescribeStateMachine.html) that are not included in the resource schema are: `creationDate` and `status`. These are not included in the resource schema since they are not included in the response from [CreateStateMachine](https://docs.aws.amazon.com/step-functions/latest/apireference/API_CreateStateMachine.html).